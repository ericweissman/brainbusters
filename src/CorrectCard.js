import React, { Component } from 'react';
import './styles/Main.scss';

class CorrectCard extends Component {
  render() {
    console.log(this.props.id)
    return(
      <div className="correct-card">
        <h4>Correct!</h4>
        <p>{this.props.question}</p>
        <p>{this.props.correct_answer}</p>
        <button onClick={() => this.props.updateGuessedCards(this.props.id)}>Back to Quiz</button>      
      </div>
    )
  }
}

export default CorrectCard;
