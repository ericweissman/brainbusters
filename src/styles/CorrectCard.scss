.correct-card {
  width: 500px;
  border-radius: 15px;
  border: 2px solid $dark;
  margin: 20px;
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: $dark;
  color: whitesmoke;

  h4 {
    font-family: $title;
    font-size: 40px;
    margin: 5px auto;
  }
  
  p {
    width: 285px;
    text-align: center;
    font-family: $text;
    font-weight: 800;
    margin: 5px auto;
  }

  button {
    background-color: $white;
    border-radius: 15px;
    padding: 5px;
    width: 100px;
    margin: 10px;
    &:hover {
        background: linear-gradient(to right, #E100FF, #7F00FF);
        color: $white;
        cursor: pointer;
    }
  }
}

  .correct-answer {
    color: $green;
    font-weight: 800;
    font-size: 25px;
  }