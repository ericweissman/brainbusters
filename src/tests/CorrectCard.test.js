import React from 'react';
import CorrectCard from '../CorrectCard.js'
import setupTests from '../setupTests'
import { shallow } from 'enzyme';

const question = {
  "id": 1,
  "question": "What array prototype method would return a sum of the elements in an array?",
  "answers": [
    ".sort()",
    ".reduce()",
    ".forEach()"
  ],
  "correct_answer": ".reduce()",
  "concept": "prototypes"
}
const correct_answer = "Testing is my favorite hobby."
const id = 21
const updateGuessedCardsMock = jest.fn()
const showAllCardsMock = jest.fn()
const updateStudyListMock = jest.fn()

describe('CorrectCard', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <CorrectCard 
          question={question}
          correct_answer={correct_answer}
          id={id}
          updateGuessedCards={updateGuessedCardsMock}
          showAllCards={showAllCardsMock}
          updateStudyList={updateStudyListMock}
      />
    )
  });

  it('should match snapshot when all data is passed correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call updateStudyList when the back-to-quiz-btn is clicked', () => {
    wrapper.find('.back-to-quiz-btn').simulate('click');
    expect(updateStudyListMock).toBeCalled();
  })


});