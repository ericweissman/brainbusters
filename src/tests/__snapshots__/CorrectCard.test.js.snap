// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CorrectCard should match snapshot when all data is passed correctly 1`] = `
<div
  className="correct-card"
>
  <h4>
    Correct!
  </h4>
  <img
    src="https://media.giphy.com/media/PS7d4tm1Hq6Sk/giphy.gif"
  />
  <p>
    <Component />
  </p>
  <p
    className="correct-answer"
  >
    Testing is my favorite hobby.
  </p>
  <button
    className="back-to-quiz-btn"
    onClick={[Function]}
  >
    Back to Quiz
  </button>
</div>
`;
